### PR TITLE
Implement mob detection!

### PR DESCRIPTION
**Pull request by stormdirus2**
It is based around the following events:

* Attacking a hostile mob (including Angerable)

* Getting attacked by a hostile mob (including Angerable)

* Colliding with a hostile mob (excludes Angerable)

When one of the following events happens, non-Angerable hostile entities within a line of sight with the player's host will begin to target them.

### CODE REVIEW

**Comment by stormdirus2** [comment]
Might wanna review the code, I am known to be a bit messy.

File: requiem-core/src/main/java/ladysnake/requiem/core/util/DetectionHelper.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.List;
+
+/**
+ * Simple mob detection helper thingy
+ *
+ * @author SciRave
+ */
+public class DetectionHelper {
+
+    //Controls what is defined as a valid enemy for the system to anger.
```
Comment (Pyrofab) [inline requiem-core/src/main/java/ladysnake/requiem/core/util/DetectionHelper.java]: Why not make this a javadoc comment ?

Comment (stormdirus2) [reply requiem-core/src/main/java/ladysnake/requiem/core/util/DetectionHelper.java]: No reason not to, I was just working as fast as possible lol. Cause' I knew I could get it up and working in a pretty fast time frame.

Comment (stormdirus2) [reply requiem-core/src/main/java/ladysnake/requiem/core/util/DetectionHelper.java]: Here, does this look better?

**Review by Pyrofab — APPROVED** [review]
Fantastic, thank you very much ! Should help balance possession's stealth a little bit.

**Comment by James103** [comment]
Could that be changed via a config option? Not all players want nearby mobs becoming hostile to you just because you were possessing a mob and you interacted with another mob.

**Comment by Pyrofab** [comment]
We are currently in alpha, so we'd rather gather feedback to improve this feature. A gamerule could definitely be added at a later point.

**Pull request merged**